### PR TITLE
add set_labels to clean_names()

### PR DESCRIPTION
**Pull request by jospueyo**
Implements #563.

I added the `set_labels` argument ensuring backward compatibility for default (not for dimnames) and sf methods. I'm not confident enough with tidygraph to implement the method for that.

I did not create labels when dimnames is used because I thought when dimnames is used, `dat` is probably a matrix, which doesn't accept column labels, am I right?

I tested for both methods.

I added the new feature in NEWS and myself as a contributor in the DESCRIPTION (thanks for this).

Also, I fixed a note regarding a bad header in NEWS (line 27).

### CODE REVIEW

File: R/clean_names.R
Comment (billdenney) [inline R/clean_names.R:69]: Please make `set_labels` an argument to the methods where it applies rather than the generic.  There may be some scenarios where `set_labels` does not make sense.

File: R/clean_names.R
Comment (billdenney) [inline R/clean_names.R:27]: Please modify to "... in each column of the returned data.frame."

File: R/clean_names.R
Comment (billdenney) [inline R/clean_names.R:87]: Please reformat the code so that the `attr()<-` line is a separate line with braces (`{}`) around it.

File: R/clean_names.R
Comment (billdenney) [inline R/clean_names.R:97]: @sfirke, I don't know enough about `sf` objects.  Do you know if adding labels may cause any issues?  (In other words, should `set_labels` apply to this function?)

File: tests/testthat/test-clean-names.R
Comment (billdenney) [inline tests/testthat/test-clean-names.R:196]: Please use left arrow (`<-`) assignment instead of equal sign assignment to match the style of the rest of the package.

**Review by billdenney — CHANGES_REQUESTED** [review]
This looks good to me, overall.  I have a few minor comments and one note for @sfirke where I don't know as much about the `sf` package.

File: tests/testthat/test-clean-names.R
Comment (jospueyo) [reply tests/testthat/test-clean-names.R]: Done

File: R/clean_names.R
Comment (jospueyo) [reply R/clean_names.R:99]: As far as I know, it shouldn't affect at all. All operations done in the columns that are not the geometry-column are the same as done in a data.frame or tibble. I work often with sf objects, and I've never found a case where labels might suppose a problem.

File: R/clean_names.R
Comment (jospueyo) [reply R/clean_names.R]: Done

File: R/clean_names.R
Comment (jospueyo) [reply R/clean_names.R]: Done

File: R/clean_names.R
Comment (jospueyo) [reply R/clean_names.R]: Done

File: R/clean_names.R
Comment (sfirke) [reply R/clean_names.R:99]: I will defer to @jospueyo , I have never used sf objects and that sounds reasonable.

**Comment by billdenney** [comment]
I have two last requests:

* Please add text like "To convert the labels back to column names, see the `sjlabelled::label_to_colnames()` function." in the details section of the docs.  (See the issues discussion for the rationale.)
* Please separate the `sf` package tests into a separate `test_that()` block with `skip_if_not_installed("sf")` so that the tests will work even if `sf` is not installed.

**Comment by billdenney** [comment]
Thanks!  @sfirke , I think that this is ready to merge (if all the tests that are now running succeed).

**Review by billdenney — APPROVED** [review]

**Comment by olivroy** [comment]
I think this feature would be great! @sfirke, when you have the chance, could you take a look.

@jospueyo there are some merge conflicts right now. Would help to resolve them to help get the PR merged :)

**Comment by jospueyo** [comment]
> I think this feature would be great! @sfirke, when you have the chance, could you take a look.
> 
> @jospueyo there are some merge conflicts right now. Would help to resolve them to help get the PR merged :)

I updated the PR, there aren't any conflicts now.

**Comment by sfirke** [comment]
Sorry to be absent on this! I will review and act shortly.

**Comment by sfirke** [comment]
Thanks @jospueyo !  This will be in the version heading to CRAN soon.  I appreciate your patience and apologize for my inattention here.  This is a really nice feature.

**Comment by jospueyo** [comment]
> Thanks @jospueyo ! This will be in the version heading to CRAN soon. I appreciate your patience and apologize for my inattention here. This is a really nice feature.

I'm honored to have contributed to this wonderful package that helps me almost daily.

**Pull request merged**
